### PR TITLE
perf: do not include sources,types when transitive_sources,type already included

### PR DESCRIPTION
**Pull request by jbedard**
---

### Test plan

- Covered by existing test cases

### CODE REVIEW

**Comment by DavidZbarsky-at** [comment]
Before:
image

After:
image

**Comment by gregmagolan** [comment]
This one conflicts now

File: js/private/js_helpers.bzl [hunk 1]
```diff
@@ -275,14 +275,17 @@ def gather_files_from_js_infos(
...
         if JsInfo in target:
             js_info = target[JsInfo]
-            if include_sources:
-                files_depsets.append(js_info.sources)
-            if include_types:
-                files_depsets.append(js_info.types)
+
             if include_transitive_sources:
                 files_depsets.append(js_info.transitive_sources)
+            elif include_sources and hasattr(js_info, "sources"):
```
Comment (gregmagolan) [inline js/private/js_helpers.bzl:281]: drop the `hasattr`

File: js/private/js_helpers.bzl [hunk 2]
```diff
@@ -275,14 +275,17 @@ def gather_files_from_js_infos(
...
+                files_depsets.append(js_info.sources)
+
             if include_transitive_types:
                 files_depsets.append(js_info.transitive_types)
+            elif include_types and hasattr(js_info, "types"):
+                files_depsets.append(js_info.types)
```
Comment (gregmagolan) [inline js/private/js_helpers.bzl:287]: drop the `hasattr`

File: js/private/js_helpers.bzl (hunk 1 above)
Comment (jbedard) [reply js/private/js_helpers.bzl:281]: Opps, bad rebase. Thanks!

**Review by gregmagolan — APPROVED** [review]

**Pull request merged**